src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VersionFileCodeFlowUpdater.cs

```diff
@@ -606,34 +609,39 @@ await targetRepo.CommitAsync(
         ];
     }
 
-    private static bool AssetExcluded(string name, IReadOnlyCollection<string>? excludedAssets)
+    private static Matcher? GetExcludedAssetsMatcher(IReadOnlyCollection<string>? excludedAssets)
     {
-        if (excludedAssets == null)
+        if (excludedAssets == null || excludedAssets.Count == 0)
         {
-            return false;
+            return null;
         }
-        foreach (var excludedAsset in excludedAssets)
+        var matcher = new Matcher();
+        matcher.AddIncludePatterns(excludedAssets);
```

Comment:
Did you consider inverting this and use the `AddExcludePatterns` method?

Reply:
I did. It mainly helped cases where I could pass the `IEnumerable` of input assets in and get the non-excluded assets out of that. However, Matcher operates over `string` and `IEnumerable<string>`, and we have a bunch of usages that actually want to filter out `DependencyDetail` items, rather than just a list of strings. So it became super awkward to use. In the end, I landed on using IncludePatterns and just a boolean on whether there is a match